### PR TITLE
Claims academic year filter

### PR DESCRIPTION
**Pull request by Kizr**
## Context

We want to be able to see which Academic year each claim was made against.

We also want to be able to filter claims by academic year in the support console.

## Changes proposed in this pull request

- [x] Delegate `:name` to `AcademicYear` from a claim
- [x] Update the claim show page for support to display the academic year
- [x] Update the claim card for the index page to display the academic year
- [x] Add a new `order_by_date` scope to the `AcademicYear` to ensure correct order for filter display
- [x] Add an academic year condition to the claims query
- [x] Add academic_year_ids to the claims filter form
- [x] Wrote a system spec to cover the functionality

## Guidance to review

- Log in as Colin
- Navigate to the claims in the navigation
- Filter claims by the current academic year (2024 to 2025) 
- All claims should be displayed
- Clear all filters
- Filter claims by the next academic year (2025 to 2026)
- No claims should be displayed
- Clear the filter using the 2025 to 2026 pill
- All claims should be displayed

## Link to Trello card

[How we identify which claims are associated with each academic year / claim window (support console)](https://trello.com/c/a3HYWhwk/793-how-we-identify-which-claims-are-associated-with-each-academic-year-claim-window-support-console)

## Screenshots

image
image
image
image

### CODE REVIEW

File: app/assets/stylesheets/components/claim/card.scss
Comment (Nitemaeric) [inline app/assets/stylesheets/components/claim/card.scss:30]: 👍

File: app/queries/claims/claims_query.rb
Comment (Nitemaeric) [inline app/queries/claims/claims_query.rb:57]: I believe this works too. I'm not sure which is objectively better though.

A join is great with lesser records, but a subquery also works well.

```suggestion
  def academic_year_condition(scope)
    return scope if params[:academic_year_ids].blank?

    scope.joins(:academic_year).where(academic_years: { id: params[:academic_year_ids] })
  end
```

**Review by Nitemaeric — APPROVED** [review]

**Review by Nitemaeric — CHANGES_REQUESTED** [review]
I've been given some design feedback to hold this off and make tweaks before we ship this.

**Comment by Nitemaeric** [comment]
Additional requirements:

- Only show Academic Years from the list of Claim Windows.
  - `AcademicYear.where(id: Claims::ClaimWindow.select(:academic_year_id)`
- Only show the Academic Year filter if there is more than 1 option.
- Add a read-only Academic Year field to the Claim creation check page under the School name.
- Add a read-only Academic Year field to the Claim details page under the School name on the User's side too.

**Comment by Kizr** [comment]
> @Kizr just FYI, I think you've got your commit messages the wrong way around here. These commits do a different thing compared to their commit messages.
> 
> * [Select academic years from claim windows](https://github.com/DFE-Digital/itt-mentor-services/pull/1024/commits/17536b95c09dcab1dacd8b41f1c178aba7e9918c) ([17536b9](https://github.com/DFE-Digital/itt-mentor-services/commit/17536b95c09dcab1dacd8b41f1c178aba7e9918c))
> * [Only show academic year filter when more than one academic years exist](https://github.com/DFE-Digital/itt-mentor-services/pull/1024/commits/735a14e8ef63a951b6f3521f7a7a8b59ed99a4c0) ([735a14e](https://github.com/DFE-Digital/itt-mentor-services/commit/735a14e8ef63a951b6f3521f7a7a8b59ed99a4c0))
> 
> They need swapping around 😄

I have no idea how I managed to do that 😕

**Review by ollietreend — APPROVED** [review]

File: app/views/claims/support/claims/_filter.html.erb
Comment (Nitemaeric) [inline app/views/claims/support/claims/_filter.html.erb:123]: TIL `#many?` ❤️

**Review by Nitemaeric — APPROVED** [review]
Nice one! Thanks!

Product would like to review this one in a review app before we merge though.

**Pull request merged**
